metricflow/cli/cli_context.py

```diff
@@ -1,23 +1,17 @@
 from __future__ import annotations
 
 import logging
-from dataclasses import dataclass
```

Comment:
Can we get a "why is this commit here" update to the commit message? It's frustrating to look through git log and see commit messages like "Removed x" or "Updated Y" and then see something odd and have to wonder "why the heck did we do that?"

Also, it's not at all clear to me why this commit is part of this PR. We *should* be removing this code (and I've also done so locally but haven't shipped the stack out for review yet), but it kind of belongs at top level tagged to the relevant issues.

Reply:
Context is that this code was going to be removed at some point anyway, and by removing it, it the reduced the dependencies that made the rebase easier, or it enabled more use of protocols vs Pydantic implementations.

Reply:
Updated commit message.